matter_server/common/helpers/util.py

```diff
@@ -291,3 +292,33 @@ def chip_core_version() -> str:
         # TODO: Fix this once we can install our own wheels on macos.
         return chip_clusters_version()
     return package_version(CHIP_CORE_PKG_NAME)
+
+
+def convert_hex_string(hex_str: str | bytes) -> str:
+    """Convert hex string received from the sdk to a regular (unicode) string."""
+    if isinstance(hex_str, str):
+        # note that the bytes string can be optionally base64 encoded
+        # when we send it back and forth over our api
+        hex_str = base64.b64decode(hex_str)
+
+    return "".join(f"{byte:02x}" for byte in hex_str)
+
+
+def convert_mac_address(hex_mac: str | bytes) -> str:
+    """Convert hexadecimal MAC received from the sdk to a regular mac-address string."""
+    if isinstance(hex_mac, str):
+        # note that the bytes string can be optionally base64 encoded
+        hex_mac = base64.b64decode(hex_mac)
+
+    return ":".join("{:02x}".format(byte) for byte in hex_mac)  # pylint: disable=C0209
+
+
+def convert_ip_address(hex_ip: str | bytes, ipv6: bool = False) -> str:
+    """Convert hexadecimal IP received from the sdk to a regular IP string."""
```

Comment:
Nit
```suggestion
    """Convert Base64 encoded byte array IP received from the Matter SDK to a regular IP string."""
```

Reply:
well, its only base64 encoded if we've sent it over the wire (as we do turn bytes into base64 strings) so the docstring was a bit confusing :-)

Reply:
From what I can tell, also the input of this function is base64, e.g. from the test you call it with:

```
convert_ip_address("KgARtxIxhABW70T//kmvxg==", True)
                    ^ this is the IP address represented as byte array encoded as a base64 string.
```

Reply:
it accepts both - if its a string, its base64 encoded

Reply:
Note that we convert it to a base64 string when sending it over the API as we cant send byte sin json

Reply:
Right :+1:  LGTM!